glass/observations.py

```diff
@@ -328,3 +330,192 @@ def tomo_nz_gausserr(
+    vardepth_map : array_like
+        Map of variable which traces the depth per tomographic bin. If vardepth_tomo_functions is not provided, the values are treated like a map of galaxy count ratios.
+    n_bins : int
+        Number of tomographic bins.
+    zbins : array_like
+        Shell redshift limits.
+    ztomo : array_like, optional
+        Tomographic redshift bin limits.
+    dndz : array_like, optional
+        Redshift distributions per tomographic bin.
+    z : array_like, optional
+        Redshift domain of dndz.
+    dndz_vardepth : array_like
+        Redshift distribution affected by variable depth (n_bins x len(vardepth_values) x len(z)).
+    vardepth_values : array_like
+        Variable depth tracer domain/values of dndz_vardepth.
+    vardepth_los_tracer : array_like, optional
+        Map of the variable depth tracer for line-of-sight direction. If provided, it is assumed to cover the same domain as vardepth_values.
+    vardepth_tomo_functions : array_like, optional
+        List of functions which map the input vardepth_map to the ratio between the galaxy count due to the variable depth and the galaxy count without variable depth (for each tomographic bin). If provided, it is assumed that there is one vardepth_map which traces the variable depth for all tomographic bins.
+
+    References
+    ----------
+    .. [1] Joachimi B., Lin, C.-A., et al., 2021, A&A, 646, A129.
+           doi:10.1051/0004-6361/202038831
+    .. [2] von Wietesheim-Kramsta M., Lin, K., et al., 2024, A&A, 695, A223.
+           doi:10.1051/0004-6361/202450487
+    """
+
+    def __init__(
+        self,
+        vardepth_map,
+        n_bins,
+        zbins,
+        ztomo,
+        dndz,
+        z,
+        dndz_vardepth,
+        vardepth_values,
+        vardepth_los_tracer=None,
+        vardepth_tomo_functions=None,
+    ):
+        super().__init__(vardepth_map, n_bins, zbins)
+        self.ztomo = ztomo
+        self.dndz = dndz
+        self.z = z
+        self.dndz_vardepth = dndz_vardepth
+        self.vardepth_values = vardepth_values
+        self.vardepth_los_tracer = vardepth_los_tracer
+        self.vardepth_tomo_functions = vardepth_tomo_functions
+
+        if vardepth_tomo_functions is not None:
+            self.vardepth_map = np.atleast_2d(vardepth_map).reshape(1, -1)
+
+    def get_los_fraction(self, index):
+        r"""Gets the fraction of galaxies affected by variable depth in the line-of-sight direction for the given tomographic bin and shell.
+
+        Parameters
+        ----------
+        index : tuple
+            Indices of the tomographic bin and shell pair.
+
+        Returns
+        -------
+        fraction_vardepth : array_like
+            Fraction of galaxies affected by variable depth in the line-of-sight direction.
+        """
+        is_in_shell = (self.zbins[index[1]][0] < self.z) & (
+            self.z <= self.zbins[index[1]][1]
+        )
+
+        n_gal_in_tomo_vardepth = np.trapz(
```

Comment:
We are now using NumPy v2, so `np.trapz` is deprecated, please change to 

```suggestion
        n_gal_in_tomo_vardepth = np.trapezoid(
```